apps/coscrad-frontend/src/app/build-routes.tsx

```diff
@@ -0,0 +1,123 @@
+import { CategorizableType, isResourceType } from '@coscrad/api-interfaces';
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { About } from '../components/about/about';
+import { Credits } from '../components/credits/credits';
+import { Home } from '../components/home/home';
+import { ListenLivePage } from '../components/listen-live-page/listen-live-page';
+import { NoteIndexContainer } from '../components/notes/note-index.container';
+import { ResourceInfoContainer } from '../components/resource-info/resource-info.container';
+import { TagDetailContainer } from '../components/tags/tag-detail.container';
+import { TagIndexContainer } from '../components/tags/tag-index.container';
+import { CategoryTreeContainer } from '../components/tree-of-knowledge/category-tree.container';
+import { ConfigurableContent } from '../configurable-front-matter/data/configurable-content-schema';
+import { contentConfig } from '../configurable-front-matter/data/content.config';
+import { bootstrapIndexToDetailFlowRoutes } from './bootstrap-index-to-detail-flow-routes';
+import { routes } from './routes/routes';
+
+export type CoscradRoute = {
+    path: string;
+    element: JSX.Element;
+    children?: CoscradRoute[];
+};
+
+export const buildRoutes = ({
+    indexToDetailFlows,
+    shouldEnableWebOfKnowledgeForResources,
+    listenLive,
+}: ConfigurableContent): CoscradRoute[] => {
+    const noteIndexToDetailConfig = indexToDetailFlows.find(
+        ({ categorizableType }) => categorizableType === CategorizableType.note
+    );
+
+    const dynamicRoutes: CoscradRoute[] = [
+        ...(!isNullOrUndefined(noteIndexToDetailConfig) && shouldEnableWebOfKnowledgeForResources
```

Comment:
Could these be 'if statements' ?

Reply:
This is a ternary. Effectively, it is acting as an inline `if...else`. The magic is to combine this with the spread operator (...) and to make the else `[]` so that there is no effect on the outcome if the check fails.  I can look at if there is a more readable way to do this, though.